src/arrayMethodJoin.js

```diff
@@ -5,8 +5,30 @@
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(str) {
+    let string = '';
+    let separator = str;
+    if (separator === undefined) {
+      separator = ',';
```

Comment:
use function default parametres
https://developer.mozilla.org/uk/docs/Web/JavaScript/Reference/Functions/Default_parameters

Then you could initialize const separator without if

Reply:
1

Reply:
I have to use let
`let separator = str;
    if (separator === null) {
      separator = 'null';
    }`
for passing the test
`test(`join with null`, () => {
  expect(source.join2(null))
    .toBe('0null1null2null3');
});`